From 54d3e89da103d7925b15df232fab5fc41844f0b0 Mon Sep 17 00:00:00 2001
From: Andreas Henriksson <andreas@fatal.se>
Date: Tue, 20 Jul 2010 11:33:45 +0200
Subject: [PATCH 7/7] Patch from upstream (rev 2537)

Replaces previous patch "0007-Ignore-ENOSYS-error-when-sett..."
with upstream commit fixing the same issue.

"Ignore ENOSYS error when setting up xattrs. (Closes: #588925)"

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=588925#10
---
 libarchive/archive_read_disk_entry_from_file.c |    2 +-
 libarchive/archive_write_disk.c                |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libarchive/archive_read_disk_entry_from_file.c b/libarchive/archive_read_disk_entry_from_file.c
index 27671df..c49e755 100644
--- a/libarchive/archive_read_disk_entry_from_file.c
+++ b/libarchive/archive_read_disk_entry_from_file.c
@@ -398,7 +398,7 @@ setup_xattrs(struct archive_read_disk *a,
 		list_size = listxattr(path, NULL, 0);
 
 	if (list_size == -1) {
-		if (errno == ENOTSUP)
+		if (errno == ENOTSUP || errno == ENOSYS)
 			return (ARCHIVE_OK);
 		archive_set_error(&a->archive, errno,
 			"Couldn't list extended attributes");
diff --git a/libarchive/archive_write_disk.c b/libarchive/archive_write_disk.c
index 60699e0..8f5d868 100644
--- a/libarchive/archive_write_disk.c
+++ b/libarchive/archive_write_disk.c
@@ -2456,7 +2456,7 @@ set_xattrs(struct archive_write_disk *a)
 				    name, value, size, 0);
 			}
 			if (e == -1) {
-				if (errno == ENOTSUP) {
+				if (errno == ENOTSUP || errno == ENOSYS) {
 					if (!warning_done) {
 						warning_done = 1;
 						archive_set_error(&a->archive, errno,
@@ -2523,7 +2523,7 @@ set_xattrs(struct archive_write_disk *a)
 				    namespace, name, value, size);
 			}
 			if (e != (int)size) {
-				if (errno == ENOTSUP) {
+				if (errno == ENOTSUP || errno == ENOSYS) {
 					if (!warning_done) {
 						warning_done = 1;
 						archive_set_error(&a->archive, errno,
-- 
1.7.1

